contracts/ERC20MultiDelegate.sol

```diff
@@ -173,9 +191,63 @@ contract ERC20MultiDelegate is ERC1155, Ownable {
         require(token.transferFrom(proxyAddressFrom, msg.sender, amount));
     }
 
-    function setUri(string memory uri) external onlyOwner {
-        _setURI(uri);
-        emit MetadataURIUpdated(uri);
+    function setUri(string memory /*uri*/) external view onlyOwner {
+        revert NotImplemented();
+    }
+
+    function tokenURI(uint256 tokenId) public view returns (string memory) {
+        string memory addressString = Strings.toHexString(uint160(tokenId), 20);
+        string memory reversedName = string.concat(
+            this.trim(addressString, 2, bytes(addressString).length),
+            ".addr.reverse"
+        );
+        (bytes memory encodedReversedName, ) = reversedName.dnsEncodeName();
+
+        string memory resolvedName;
+        try metadataResolver.reverse(encodedReversedName) returns (
+            string memory _resolvedName,
+            address,
+            address,
+            address
+        ) {
+            resolvedName = _resolvedName;
+        } catch {
+            return "";
+        }
+
+        (bytes memory encodedName, bytes32 namehash) = resolvedName
+            .dnsEncodeName();
+        bytes memory data = abi.encodeWithSignature(
+            "text(bytes32,string)",
+            [namehash, "avatar"]
+        );
+        (bytes memory result, ) = metadataResolver.resolve(encodedName, data);
+        string memory imageUri = result.length == 0
+            ? ""
+            : abi.decode(result, (string));
+        string memory json = Base64.encode(
+            bytes(
+                string.concat(
+                    '{"name": "',
+                    resolvedName,
+                    " Delegate Token",
+                    '", "token_id": "',
+                    Strings.toString(tokenId),
+                    '", "description": "This NFT is a proof for your ENS delegation strategy.", "image": "',
+                    imageUri,
+                    '"}'
+                )
+            )
+        );
+        return string.concat("data:application/json;base64,", json);
+    }
+
+    function trim(
```

Comment:
![009ea33df7aca6f9c65c51ae10df081d](https://github.com/ensdomains/governance-contracts/assets/17865/680cf303-0ea7-4fee-b9df-8c04de146d4e)

Copying the address-to-hex implementation into /utils/HexUtils.sol, and removing the 0x prefix, would remove the need for this function.

Reply:
added a new HexUtils method, which converts address to hex string without hex prefix. ptal if that's okay.